configure.php

```diff
@@ -0,0 +1,528 @@
+#!/usr/bin/env php
+<?php
+/**
+ * Configure the alleyinteractive/create-wordpress-project interactively.
+ *
+ * This script will:
+ *
+ *   1. Prompt the user to replace the default values in the project.
+ *   2. Configure it for the project's hosting provider (VIP, Pantheon, etc.)
+ *   3. Scaffold default plugin for the project from alleyinteractive/create-wordpress-plugin.
+ *
+ * Supports arguments to set the values directly.
+ *
+ * [--project_name=<project_name>]
+ * : The project name.
+ *
+ * [--author_name=<author_name>]
+ * : The author name.
+ *
+ * [--author_email=<author_email>]
+ * : The author email.
+ *
+ * phpcs:disable
+ */
+
+namespace Create_WordPress_Project\Configure;
+
+if ( ! defined( 'STDIN' ) ) {
+	die( 'Not in CLI mode.' );
+}
+
+if ( 0 === strpos( strtoupper( PHP_OS ), 'WIN' ) ) {
+	die( 'Not supported in Windows. 🪟' );
+}
+
+if ( version_compare( PHP_VERSION, '8.2.0', '<' ) ) {
+	die( 'PHP 8.2.0 or greater is required.' );
+}
+
+// Parse the command line arguments from $argv.
+$args         = [];
+$previous_key = null;
+
+foreach ( $argv as $value ) {
+	if ( str_starts_with( $value, '--' ) ) {
+		if ( false !== strpos( $value, '=' ) ) {
+			[ $arg, $value ] = explode( '=', substr( $value, 2 ), 2 );
+
+			$args[ $arg ] = trim( $value );
+
+			$previous_key = null;
+		} else {
+			$args[ substr( $value, 2 ) ] = true;
+
+			$previous_key = substr( $value, 2 );
+		}
+	} elseif ( ! empty( $previous_key ) ) {
+		$args[ $previous_key ] = trim( $value );
+	} else {
+		$previous_key = trim( $value );
+	}
+}
+
+$terminal_width = (int) exec( 'tput cols' );
+
+function write( string $text ): void {
+	global $terminal_width;
+	echo wordwrap( $text, $terminal_width - 1 ) . PHP_EOL;
+}
+
+function ask( string $question, string $default = '', bool $allow_empty = true ): string {
+	while ( true ) {
+		write( $question . ( $default ? " [{$default}]" : '' ) . ': ' );
+		$answer = readline( '> ' );
+
+		$value = $answer ?: $default;
+
+		if ( ! $allow_empty && empty( $value ) ) {
+			echo "This value can't be empty." . PHP_EOL;
+			continue;
+		}
+
+		return $value;
+	}
+}
+
+function confirm( string $question, bool $default = false ): bool {
+	write( "{$question} (yes/no) [" . ( $default ? 'yes' : 'no' ) . ']: ' );
+
+	$answer = readline( '> ' );
+
+	if ( ! $answer ) {
+		return $default;
+	}
+
+	return in_array( strtolower( trim( $answer ) ), [ 'y', 'yes', 'true', '1' ], true );
+}
+
+function run( string $command, string $dir = null ): string {
+	$command = $dir ? "cd {$dir} && {$command}" : $command;
+
+	return trim( (string) shell_exec( $command ) );
+}
+
+function str_after( string $subject, string $search ): string {
+	$pos = strrpos( $subject, $search );
+
+	if ( $pos === false ) {
+		return $subject;
+	}
+
+	return substr( $subject, $pos + strlen( $search ) );
+}
+
+function slugify( string $subject ): string {
+	return strtolower( trim( (string) preg_replace( '/[^A-Za-z0-9-]+/', '-', $subject ), '-' ) );
```

Comment:
Could be simplified by changing the order of operations. `strtolower` could be run first, followed by `preg_replace`, which would only need to check for lowercase letters, followed by `trim`.